test/commands.test.ts

```diff
@@ -0,0 +1,259 @@
+        ];
+
+        await inTextEditor({ language: 'org', content: steps[0] }, async (e, d) => {
+            for (let i = 1; i < steps.length; ++i) {
+                await vscode.commands.executeCommand('org.doPromote');
+                assert.equal(d.getText(), steps[i]);
+            }
+        });
+    });
+
+    test('Bold', async () => {
+        const initial = 'Some text here';
+        const expected = 'Some *text* here';
+        const textWordRange = new vscode.Range(0, 5, 0, 9);
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            select(e, textWordRange);
+            await vscode.commands.executeCommand('org.bold');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('Italic', async () => {
+        const initial = 'Some text here';
+        const expected = 'Some /text/ here';
+        const textWordRange = new vscode.Range(0, 5, 0, 9);
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            select(e, textWordRange);
+            await vscode.commands.executeCommand('org.italic');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('Underline', async () => {
+        const initial = 'Some text here';
+        const expected = 'Some _text_ here';
+        const textWordRange = new vscode.Range(0, 5, 0, 9);
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            select(e, textWordRange);
+            await vscode.commands.executeCommand('org.underline');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('Code', async () => {
+        const initial = 'Some text here';
+        const expected = 'Some ~text~ here';
+        const textWordRange = new vscode.Range(0, 5, 0, 9);
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            select(e, textWordRange);
+            await vscode.commands.executeCommand('org.code');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('Verbose', async () => {
+        const initial = 'Some text here';
+        const expected = 'Some =text= here';
+        const textWordRange = new vscode.Range(0, 5, 0, 9);
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            select(e, textWordRange);
+            await vscode.commands.executeCommand('org.verbose');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('Literal', async () => {
+        const initial = 'Some text here';
+        const expected = ': Some text here';
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            await vscode.commands.executeCommand('org.literal');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('InsertSubheading', async () => {
+        const initial = `* Header
```

Comment:
why using backquotes?

Reply:
I find backquoted string much more readable when working with multiline content. However, somewhere I need to use concatenated strings, because VSCode strips trailing whitespaces in backquoted strings which leads to incorrect data.

Reply:
okey dokey